### PR TITLE
expression: provide `OptPropPrivilegeChecker` for `EvalContext`

### LINKED ISSUE #56301: provide `OptPropPrivilegeChecker` for `EvalContext`

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #56301

Problem Summary:

It's better to make the privilege check an "optional" property in `EvalContext` for reasons:

- most built-in functions do not need to check privileges. Force to provide privilege checking for all cases is not easy to use.
- For security. In the previous implementation, the privilege check allows all privileges by default in `exprstatic.EvalContext`. There is no problem now because all scenes which are using these static contexts disallow these functions that require privilege. However, this may introduce potential security risks in the future scenes if we allows some dangerous function but forgot to set the check privilege methods. This PR forces the functions that need privilege check to require `OptPropPrivilegeChecker` property to avoid it.

### What changed and how does it work?

- Provide `OptPropPrivilegeChecker` for `EvalContext`
- Remove common privilege check functions in `EvalContext`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]
LGTM

File: pkg/expression/builtin_info.go [hunk 1]
```diff
@@ -1217,6 +1243,15 @@ func (b *builtinTiDBDecodeSQLDigestsSig) Clone() builtinFunc {
 }
 
 func (b *builtinTiDBDecodeSQLDigestsSig) evalString(ctx EvalContext, row chunk.Row) (string, bool, error) {
+	privChecker, err := b.GetPrivilegeChecker(ctx)
+	if err != nil {
+		return "", true, err
+	}
+
+	if !privChecker.RequestVerification("", "", "", mysql.ProcessPriv) {
```
[the remark this answers is not in the document]
Comment (lcwangchao) [reply pkg/expression/builtin_info.go:1251]: This checks `process_priv` which is used to check some privileges for `show processlist`. But seems we extended it in some special functions. This PR just follows the previous implementation here: https://github.com/pingcap/tidb/pull/56302/files#diff-bcd6b5e9b7f2ab3b56ccd7cec28822f1ef3825f3bf9abdf4b8199765e5cd7ce3L1183

**Review by XuHuaiyu — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**